### DISCUSSION TITLE
How to delete a parquet file from a existing context?

### DISCUSSION THREAD

**Discussion by Ma1oneZhang** [apache/datafusion#9979] [Q&A]
## Background
 I'm currently engaged in a project utilizing datafusion, which I find to be both convenient and efficient for database project development. However, I've encountered a confusing issue: once a Parquet file is registered to the SessionContext, it seems there's no supported method to deregister it. Consequently, if the file is deleted remotely, a NotFound Error is reported. Could anyone advise on how to deregister a Parquet file from a SessionContext?

**Comment by haohuaijin** [comment]
maybe `SessionContext` 's `deregister_table` function can do this task. because `register_parquet` underly call `register_table` function.

**Reply by Ma1oneZhang** [reply]
The `deregister_table` function will detach the specified table from the schema; however, my intention is not to delete all the files contained within the table, but rather to selectively remove only certain ones.

**Comment by alamb** [comment]
The only way i know of to do this today is to drop the table and recreate it (which will re-scan all parquet files)
